File: Selenium/Activities/Activity13.py
import win32clipboard
from selenium import webdriver
from selenium.webdriver import ActionChains, Keys
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

service = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=service) as driver:

    driver.get("https://training-support.net/selenium/tables")

    print("Title of Tables page: ", driver.title)

    columns = driver.find_elements(By.XPATH,
                "//table[contains(@class, 'strip')]/tbody/tr[1]/td")
    print("Number of columns: ", len(columns))

    rows = driver.find_elements(By.XPATH,
                        "//table[contains(@class, 'strip')]/tbody/tr")
    print("Number of rows: ", len(rows))

    third_row = driver.find_elements(By.XPATH,
        "//table[contains(@class, 'strip')]/tbody/tr[3]/td")
    print("Cell values of third row:")
    for row in third_row:
        print("Cell value: ", row.text)

    second_row_second_cell = driver.find_element(By.XPATH,
                "//table[contains(@class, 'strip')]/tbody/tr[2]/td[2]")
    print("Cell value at second row and second column: ", second_row_second_cell.text)

    driver.close()

File: Selenium/Activities/Activity13.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.awt.datatransfer.UnsupportedFlavorException;
import java.io.IOException;
import java.time.Duration;
import java.util.List;

public class Activity13 {
    public static void main(String[] args) throws InterruptedException, IOException, UnsupportedFlavorException {
        WebDriverManager.firefoxdriver().setup();

        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://www.training-support.net/selenium/tables");
        System.out.println("Title of Tables page: " + driver.getTitle());

        List<WebElement> rows = driver.findElements(By
                .xpath("//table[contains(@class, 'strip')]" +
                        "/tbody/tr"));
        System.out.println("Number of rows: " + rows.size());

        List<WebElement> cols = driver.findElements(By
                .xpath("//table[contains(@class, 'strip')]" +
                        "/tbody/tr[1]/td"));
        System.out.println("Number of columns: " + cols.size());

        List<WebElement> thirdRow = driver.findElements(By
                        .xpath("//table[contains(@class, " +
                                "'strip')]/tbody/tr[3]/td"));
        System.out.println("Third row cell values: ");
        for(WebElement cell : thirdRow) {
            System.out.println(cell.getText());
        }

        WebElement cellValue = driver.findElement(By
                .xpath("//table[contains(@class, 'strip')]" +
                        "/tbody/tr[2]/td[2]"));
        System.out.println("Second row, second cell value: " + cellValue.getText());

        driver.close();




    }
}
